src/HealthChecks.AzureServiceBus/AzureEventHubHealthCheck.cs

```diff
@@ -12,34 +12,48 @@ public class AzureEventHubHealthCheck
     {
         private static readonly ConcurrentDictionary<string, EventHubClient> _eventHubConnections = new ConcurrentDictionary<string, EventHubClient>();
 
-        private readonly string _eventHubConnectionString;
+        public string EntityPath => ConnectionStringBuilder.EntityPath;
+        public Uri Endpoint => ConnectionStringBuilder.Endpoint;
+        protected EventHubsConnectionStringBuilder ConnectionStringBuilder { get; }
+
         public AzureEventHubHealthCheck(string connectionString, string eventHubName)
         {
             if (string.IsNullOrEmpty(connectionString))
             {
                 throw new ArgumentNullException(nameof(connectionString));
-            }
-
+            }
+
             if (string.IsNullOrEmpty(eventHubName))
             {
                 throw new ArgumentNullException(nameof(eventHubName));
             }
 
-            var builder = new EventHubsConnectionStringBuilder(connectionString)
+            ConnectionStringBuilder = new EventHubsConnectionStringBuilder(connectionString)
             {
                 EntityPath = eventHubName
             };
-            _eventHubConnectionString = builder.ToString();
         }
+
+        public AzureEventHubHealthCheck(string connectionString)
```

Comment:
Why you need this new constructor

Reply:
To make `eventHubName` optional given it can be part of the connection string, see other comment & https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/issues/469